packages/sirius-web/frontend/sirius-web-application/src/views/edit-project/workbench-views/explorer/ExplorerView.tsx

```diff
@@ -120,6 +148,96 @@ export const ExplorerView = ({ editingContextId, readOnly }: WorkbenchViewCompon
     return null;
   }, [treeElement]);
 
+  const { getTreePath, loading: treePathLoading, data: treePathData } = useTreePath();
+  const {
+    getExpandAllTreePath,
+    loading: expandAllTreePathLoading,
+    data: expandAllTreePathData,
+  } = useExpandAllTreePath();
+
+  // If we should auto-expand to reveal the selection, we need to compute the tree path to expand
+  const selectionKey: string = selection?.entries
+    .map((entry) => entry.id)
+    .sort()
+    .join(':');
+  useEffect(() => {
+    if (state.synchronizedWithSelection) {
+      if (state.tree?.id) {
+        const variables: GQLGetTreePathVariables = {
+          editingContextId,
+          treeId: state.tree.id,
+          selectionEntryIds: selection.entries.map((entry) => entry.id),
+        };
+        getTreePath({ variables });
+      }
+    }
+  }, [editingContextId, selectionKey, state.synchronizedWithSelection, getTreePath]);
+
+  useEffect(() => {
+    if (!treePathLoading) {
```

Comment:
We can remove at least two "if" from this function without a change in behavior, it would improve the readability

Reply:
Done